"the method rev() exists on usize" is misleading

<!--
Thank you for filing a bug report! 🐛 Please provide a short summary of the bug,
along with any information you feel relevant to replicating the bug.

If you cannot produce a minimal reproduction case (something that would work in
isolation), please provide the steps or even link to a repository that causes
the problematic output to occur.
-->

Given the following code: <!-- Please provide a link to play.rust-lang.org -->

```rust
  let arr = &[0,1,2,3];
  for i in 0..arr.len().rev() {
     // ...   
  }
```

The current output is:

```
error[E0599]: the method `rev` exists for type `usize`, but its trait bounds were not satisfied
 --> src/main.rs:5:25
  |
5 |   for i in 0..arr.len().rev() {
  |                         ^^^ method cannot be called on `usize` due to unsatisfied trait bounds
  |
  = note: the following trait bounds were not satisfied:
          `usize: Iterator`
          which is required by `&mut usize: Iterator
```

<!-- The following is not always necessary. -->
Ideally the output should look like:


```
// Minimal fix: remove misleading message:
Omit "the method rev exists for type usize." 

// Amazing fix:
  |
5 |   for i in 0..arr.len().rev() {
  |                         ^^^ method cannot be called on `usize` due to unsatisfied trait bounds
  |
  = note: the following trait bounds were not satisfied:
          `usize: Iterator`
          which is required by `&mut usize: Iterator
  = hint: (0..arr.len()) implements `Iterator`, perhaps you are missing parentheses? 

```

<!--
If the problem is not self-explanatory, please provide a rationale for the
change.
-->

<!--
If dramatically different output is caused by small changes, consider also
adding them here.

If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions. The output might also be
different depending on the Edition.
-->

--- a/src/test/ui/methods/issues/issue-90315.rs
+++ b/src/test/ui/methods/issues/issue-90315.rs
@@ -1,0 +1,7 @@
+fn main() {
+  let arr = &[0,1,2,3];
+  for _i in 0..arr.len().rev() { //~ERROR not an iterator
+     // The above error used to say “the method `rev` exists for type `usize`”.
+     // This regression test ensures it doesn't say that any more.
+  }
+}

--- a/src/test/ui/methods/issues/issue-90315.stderr
+++ b/src/test/ui/methods/issues/issue-90315.stderr
@@ -1,0 +1,13 @@
+error[E0599]: `usize` is not an iterator
+  --> $DIR/issue-90315.rs:3:26
+   |
+LL |   for _i in 0..arr.len().rev() {
+   |                          ^^^ `usize` is not an iterator
+   |
+   = note: the following trait bounds were not satisfied:
+           `usize: Iterator`
+           which is required by `&mut usize: Iterator`
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0599`.

